packages/edit-navigation/src/components/header/index.js

```diff
@@ -64,50 +70,78 @@ export default function Header( {
 					<DropdownMenu
 						icon={ null }
 						toggleProps={ {
+							children: __( 'Switch menu' ),
+							'aria-label': __(
+								'Switch menu, or create a new menu'
+							),
 							showTooltip: false,
-							children: __( 'Select menu' ),
 							isTertiary: true,
 							disabled: ! menus?.length,
 							__experimentalIsFocusable: true,
 						} }
 						popoverProps={ {
+							className:
+								'edit-navigation-header__menu-selection-dropdown',
 							position: 'bottom left',
 						} }
 					>
-						{ () => (
-							<MenuGroup>
-								<MenuItemsChoice
-									value={ selectedMenuId }
-									onSelect={ onSelectMenu }
-									choices={ menus.map( ( menu ) => ( {
-										value: menu.id,
-										label: menu.name,
-									} ) ) }
-								/>
-							</MenuGroup>
+						{ ( { onClose } ) => (
+							<>
+								<MenuGroup>
+									<MenuItemsChoice
+										value={ selectedMenuId }
+										onSelect={ ( menuId ) => {
+											onSelectMenu( menuId );
+											onClose();
+										} }
+										choices={ menus.map( ( menu ) => ( {
```

Comment:
we can use destructuring here `({id, name})`

Reply:
Sounds good, done in bf139c0